libSceVideodec2: Update structs to match newer firmwares

StevenMiller123 opened the pull request:
Based on some decompilation I did, some libSceVideodec2 structs were adjusted somewhere around firmware 6.50. Our previous code was built around older versions of the library, and as a result, newer versions of libSceAvPlayer.sprx and newer games that use libSceVideodec2 would hit the struct size error return.
This PR updates the structs, and updates the struct size checks to match how newer versions of libSceVideodec2 behave.

This should properly fix #2604, though the author of that issue probably can't test this. 

I've also thrown in a function export needed for Assassin's Creed Unity, since I don't see a point in making a separate one line PR for that.

Commit:
Update file_system.cpp

Commit:
libSceVideodec2 struct fixes

Our code was based on an old version of the libSceVideodec2 library. Based on what I've decompiled, these structs changed somewhere around firmware 6.50, and newer versions of the library have these flexible checks to accommodate both variants of the structs.

Commit:
Static assert for AvcPictureInfo struct

All the other Videodec2 structs have static asserts, might as well use one here too.

squidbus commented on src/core/libraries/videodec/videodec2.h:
```diff
@@ -73,8 +73,10 @@ struct OrbisVideodec2OutputInfo {
     void* frameBuffer;
     u64 frameBufferSize;
+    u32 frameFormat;
+    u32 framePitchInBytes;
```
Shouldn’t we also populate these with something if they’re included in the struct?

StevenMiller123 replied:
Should be fixed now, though after looking into this, I'm not sure we're setting `framePitch` properly.
I'll run some hardware tests to verify how this works once I get a chance.

Commit:
Initialize new values

Set proper values for frameFormat and framePitchInBytes.
`frame->linesize[0]` appears to be in bytes already, I'm not sure if that means framePitch is being set wrong though.
